### PR TITLE
use shapefile-js to accept shp.zip on new project areas

### PR DESCRIPTION
**Pull request by mapmeld**
when creating the project area, you can upload a zipped shapefile in addition to geojson + kml

if the zipfile is not successfully parsed as a shapefile, the pleaseProvideGeojsonOrKmlFileI18n error message is displayed

### CODE REVIEW

**Comment by pgiraud** [comment]
This is an interesting addition. I have 2 comments though.

Can you please change the texts (help and alert messages) so that it tells the user that a zipped shapefile can be provided?

There's a mismatch (not your fault) in the name of the variable used for the error message.
`between pleaseProvideGeojsonOrKmlFileI18n` should be used everywhere.
Can you please rename the variable in the following code?
https://github.com/hotosm/osm-tasking-manager2/blob/master/osmtm/templates/project.new.mako#L30

**Comment by mapmeld** [comment]
How far should I go when updating the locale?

- update POT file
- update EN PO file
- run update_catalog but only commit changes to text
- run update_catalog and include changes to line number, etc
- run update_catalog and include all updates

**Comment by pgiraud** [comment]
Please leave the gettext files untouched. We'll do the catalog updates once merged.

**Comment by mapmeld** [comment]
that makes sense.

I think this should be good

**Comment by pgiraud** [comment]
Can I ask you one last thing? Can you please squash your commits into one?

**Comment by mapmeld** [comment]
@pgiraud I think this should be good. One commit, rebased on current master

File: osmtm/templates/project.new.mako [hunk 1]
```diff
@@ -59,7 +60,7 @@
     <input type="file" val="" name="import" class="hidden" />
 <%
     link = '<a id="import" data-role="button" class="btn btn-default" rel="tooltip" title="%s">%s</a>' \
-           % (_('Provide a .geojson or .kml file.'), _('Import'))
+           % (_('Provide a .geojson, .kml, or zipped .shp file.'), _('Import'))
     text = _('${import_link} a <em>GeoJSON</em> or <em>KML</em> file.', mapping={'import_link': link})
```
Comment (pgiraud) [inline osmtm/templates/project.new.mako]: This text should also mention the shapefile option.

**Comment by pgiraud** [comment]
I gave your code a try on Chrome and Safari on mac and nothing happens.
What browser are you using?

**Comment by mapmeld** [comment]
@pgiraud did you pull in the new submodule? is shp.js loading but not being used? If so, please send me a ZIP of the shapefile

**Comment by pgiraud** [comment]
The submodule is correctly pulled.
It looks like `shapefilejs` doesn't like the shapefile I provide, though it's really simple.
http://geojson.io/#id=gist:pgiraud/7d2678d9d0d4d91042e5&map=5/10.120/12.854 (click on save as shapefile to get the corresponding shapefile).

I tried to load the zipped shapefile in this [shapefilejs demo](http://leaflet.calvinmetcalf.com/#3/32.18/10.55) and I get the following error:
```
shit RangeError: Invalid typed array length
```

**Comment by mapmeld** [comment]
@pgiraud lol you're right... I get the same error when I drop the file onto http://leaflet.calvinmetcalf.com/#3/32.69/10.55 which is the demo

@calvinmetcalf is there a way that we can predict which files can and can't be read by the script?

**Comment by calvinmetcalf** [comment]
To paraphrase Anna Karenina every working shapefile is alike, but every screwed up shapefile is screwed up in its own way.   Doing a quick debug, it looks like the issue might be in the dbf generation as the error changes if you give one of the features a property. 
 
though, to answer your question f I could predict that a shapefile wouldn't work, I'd probably be able to fix the error...

**Comment by mapmeld** [comment]
@pgiraud shpjs is now updated to accept shapefiles like yours which don't have attribute data. Good test case!

Let me know if you think we can get this merged in

**Comment by pgiraud** [comment]
Excellent.
Can you please address my pending comment?

**Comment by mapmeld** [comment]
@pgiraud the "Provide a .geojson, .kml, or zipped .shp file" label? Or something else?

**Comment by pgiraud** [comment]
Yes, the `text` variable.
https://github.com/hotosm/osm-tasking-manager2/pull/727/files#diff-beb97a9e4448c1d6e005647d61662179R64

**Comment by mapmeld** [comment]
@pgiraud ok sorry I was looking at the line above. Let me know if you have issues with this one

**Comment by ethan-nelson** [comment]
Ping #698

**Comment by pgiraud** [comment]
Thanks @mapmeld for your appreciated efforts.
Merging.

**Pull request merged**